pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -320,7 +320,25 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 			// OBSERVATION: pvc is "Pending", will retry
 			switch {
 			case delayBinding && !pvutil.IsDelayBindingProvisioning(claim):
-				ctrl.eventRecorder.Event(claim, v1.EventTypeNormal, events.WaitForFirstConsumer, "waiting for first consumer to be created before binding")
+				reason := events.WaitForFirstConsumer
```

Comment:
Please put the eventing logic into a separate method, so syncUnboundClaim contains only the important if-then-else branches.

Reply:
done